src/open_prime_rando/echoes/specific_area_patches/required_fixes.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from collections.abc import Iterable
+
+from retro_data_structures.enums.echoes import Message, State
+from retro_data_structures.formats.mrea import Area
+from retro_data_structures.formats.script_object import InstanceIdRef
+from retro_data_structures.properties.echoes.archetypes.EditorProperties import EditorProperties
+from retro_data_structures.properties.echoes.archetypes.Transform import Transform
+from retro_data_structures.properties.echoes.core.Vector import Vector
+from retro_data_structures.properties.echoes.objects import (
+    HUDMemo,
+    Pickup,
+    Relay,
+    SpawnPoint,
+    Switch,
+    Timer,
+    Trigger,
+)
+
+from open_prime_rando.echoes.asset_ids import agon_wastes, sanctuary_fortress, temple_grounds, torvus_bog
+from open_prime_rando.echoes.asset_ids.world import (
+    AGON_WASTES_MLVL,
+    SANCTUARY_FORTRESS_MLVL,
+    TEMPLE_GROUNDS_MLVL,
+    TORVUS_BOG_MLVL,
+)
+from open_prime_rando.patcher_editor import PatcherEditor
+
+LOG = logging.getLogger("echoes_patcher")
+
+
+def required_fixes(editor: PatcherEditor):
+    """
+    Applies changes necessary for the game to function properly.
+    """
+    mining_station_b(editor)
+    undertemple_access(editor)
+    main_reactor(editor)
+    sacrificial_chamber(editor)
+    aerie(editor)
+    main_research(editor)
+    hive_chamber_b(editor)
+    gfmc_compound(editor)
+    torvus_temple(editor)
+
+
+def mining_station_b(editor: PatcherEditor):
+    """
+    Normally this relay is activated by the cutscene, but has an incoming connection from the pickup.
+    Activating it like this means the pickup will trigger the relay without the cutscene
+    """
+    area = editor.get_area(AGON_WASTES_MLVL, agon_wastes.MINING_STATION_B_MREA)
+    with area.get_instance(0x80121).edit_properties(Relay) as post_pickup_relay:
+        post_pickup_relay.editor_properties.active = True
+
+
+def undertemple_access(editor: PatcherEditor):
+    """
+    Move the default spawn point in-bounds.
+    """
+    area = editor.get_area(TORVUS_BOG_MLVL, torvus_bog.UNDERTEMPLE_ACCESS_MREA)
+
+    with area.get_instance("Spawn point 001").edit_properties(SpawnPoint) as spawn:
+        spawn.editor_properties.transform.position.x = -149.25
+
+
+def main_reactor(editor: PatcherEditor):
+    """
+    Save some memory during DS1 fight.
+    """
+    area = editor.get_area(AGON_WASTES_MLVL, agon_wastes.MAIN_REACTOR_MREA)
+
+    unload_relay = area.get_instance("Unload dock when door closed")
+    trigger = area.get_instance("Trigger Start DS Intro")
+    trigger.add_connection(State.Entered, Message.SetToZero, unload_relay)
+
+
+def sacrificial_chamber(editor: PatcherEditor):
+    """
+    Unknown purpose.
```

Comment:
I remember there being a bug where the pickup only appears the first time you trigger the fight.
So you could enter the room, trigger the boss, leave, come back again, kill the boss this time and get a big nothing.

Reply:
that reminds mii - it's still possible to remove the alpha splinter pickup permanently, isn't it?

Reply:
I'd say yes. Trigger fight, jump over walls and leave.

Reply:
Yes to both Grapple Guardian and Alpha Splinter